packages/expo-updates/android/src/main/java/expo/modules/updates/UpdatesController.java

```diff
@@ -0,0 +1,512 @@
+  }
+
+  /**
+   * Returns a map of the locally downloaded assets for the current update. Keys are the remote URLs
+   * of the assets and values are local paths. This should be exported by the Updates JS module and
+   * can be used by `expo-asset` or a similar module to override React Native's asset resolution and
+   * use the locally downloaded assets.
+   */
+  public @Nullable Map<AssetEntity, String> getLocalAssetFiles() {
+    if (mLauncher == null) {
+      return null;
+    }
+    return mLauncher.getLocalAssetFiles();
+  }
+
+  // other getters
+
+  public Uri getManifestUrl() {
+    return mManifestUrl;
+  }
+
+  public File getUpdatesDirectory() {
+    return mUpdatesDirectory;
+  }
+
+  public UpdateEntity getLaunchedUpdate() {
+    return mLauncher.getLaunchedUpdate();
+  }
+
+  public SelectionPolicy getSelectionPolicy() {
+    return mSelectionPolicy;
+  }
+
+  public boolean isEmergencyLaunch() {
+    return mLauncher != null && mLauncher instanceof EmergencyLauncher;
+  }
+
+  /**
+   * Starts the update process to launch a previously-loaded update and (if configured to do so)
+   * check for a new update from the server. This method should be called as early as possible in
+   * the application's lifecycle.
+   * @param context the base context of the application, ideally a {@link ReactApplication}
+   */
+  public synchronized void start(final Context context) {
+    if (mUpdatesDirectory == null) {
+      mLauncher = new EmergencyLauncher(context, mUpdatesDirectoryException);
+      mIsReadyToLaunch = true;
+      mTimeoutFinished = true;
+      UpdatesController.this.notify();
+      return;
+    }
+
+    int delay = 0;
+    try {
+      ApplicationInfo ai = context.getPackageManager().getApplicationInfo(context.getPackageName(), PackageManager.GET_META_DATA);
+      delay = ai.metaData.getInt("expo.modules.updates.EXPO_UPDATES_LAUNCH_WAIT_MS", 0);
+    } catch (Exception e) {
+      Log.e(TAG, "Could not parse expo.modules.updates.EXPO_UPDATES_LAUNCH_WAIT_MS from AndroidManifest; defaulting to 0", e);
+    }
+
+    if (delay > 0) {
+      HandlerThread handlerThread = new HandlerThread("expo-updates-timer");
+      handlerThread.start();
+      new Handler(handlerThread.getLooper()).postDelayed(this::finishTimeout, delay);
+    } else {
+      mTimeoutFinished = true;
+    }
+
+    UpdatesDatabase database = getDatabase();
+    LauncherWithSelectionPolicy launcher = new LauncherWithSelectionPolicy(mUpdatesDirectory, mSelectionPolicy);
+    mLauncher = launcher;
+    if (mSelectionPolicy.shouldLoadNewUpdate(EmbeddedLoader.readEmbeddedManifest(context).getUpdateEntity(), launcher.getLaunchableUpdate(database))) {
+      new EmbeddedLoader(context, database, mUpdatesDirectory).loadEmbeddedUpdate();
+    }
+    launcher.launch(database, context, new Launcher.LauncherCallback() {
+      @Override
+      public void onFailure(Exception e) {
+        mLauncher = new EmergencyLauncher(context, e);
+        onSuccess();
```

Comment:
This might be extremaly misleading. Imageine someone in the future adds some code to `onSuccess` which they imagine will be executed only in case of success (i.e. analytics event). That might confuse them (or future us).